product-service/handlers/getProductById.js

```diff
@@ -0,0 +1,30 @@
+import middy from "@middy/core";
+import cors from "@middy/http-cors";
+import { autoProxyResponse } from "middy-autoproxyresponse";
+import productService from "../services/productService";
+import NotFoundError from "../common/errors/notFoundError";
+import get404Response from "../common/responses/get404Response";
+import get500Response from "../common/responses/get500Response";
+
+const handler = async event => {
+  try {
+    const { pathParameters: { id: productId } } = event;
+    const product = await productService.getProductById(productId);
+
+    if(!product) {
+      throw new NotFoundError(`The following product (id: ${productId}) is not found.`);
+    }
+
+    return product;
+  } catch (error) {
+    if(error instanceof NotFoundError) {
```

Comment:
[Proposal]

Since you are using @middy libraries it can be a good idea to extract error handling logic in a middleware